### PR TITLE
feat(.github/): Generate javadoc and host in a page branch. Fix javadoc errors in source file.

### PR DESCRIPTION
**Pull request by shouvSarker**
During testing from the git workflow, 

javadoc pushes have been made to https://github.com/agiledigital-labs/agiledigital-labs.github.io
argocd pushes have been made to https://github.com/agiledigital-labs/argo-apps/tree/test-shouv

### CODE REVIEW

**Comment by SamMurphyDev** [comment]
Are you able to put a link on the index page in the agiledigital-labs.github.io repo for the javadoc. Make life a little easier. 

```html
<ul>
  <li><a href="https://agiledigital-labs.github.io/confluence-plugin-idea-search/">Confluence Plugin Idea Search Java Doc</a></li>
</ul>

File: src/main/resources/js/TableFilter.js [hunk 1]
```diff
@@ -236,21 +236,12 @@ const calculateHiddenRowForColumn = (list) => {
     if (optionsList.length > 0) {
       const rowList = [];
 
-<<<<<<< HEAD
```
Comment (SamMurphyDev) [inline src/main/resources/js/TableFilter.js:239]: This fix is already on the develop branch. Probs need to rebase

File: src/main/resources/js/TableFilter.js [hunk 2]
```diff
@@ -4,27 +4,27 @@ const $ = AJS.$;
  * Prefix of the id uses to denote search input
  * @type {string}
  */
-const searchIdPrefix = 'search-';
```
Comment (SamMurphyDev) [inline src/main/resources/js/TableFilter.js:7]: I'm guessing a formatter went through and changed all the single quotes to double-quotes. Did Prettier do that or something else?

**Review by SamMurphyDev — COMMENTED** [review]
Couple of things otherwise looks good

File: src/main/resources/js/TableFilter.js (hunk 2 above)
Comment (shouvSarker) [reply src/main/resources/js/TableFilter.js:7]: It was done by prettier, the other javascript file `ideaSearch.js` also has all double quotes and prettier likes double quotes, so decided to keep it.

**Comment by shouvSarker** [comment]
> Are you able to put a link on the index page in the agiledigital-labs.github.io repo for the javadoc. Make life a little easier.
> 
> ```
> <ul>
>   <li><a href="https://agiledigital-labs.github.io/confluence-plugin-idea-search/">Confluence Plugin Idea Search Java Doc</a></li>
> </ul>
> ```

Added to the idex page! https://agiledigital-labs.github.io/

File: src/main/resources/js/TableFilter.js (hunk 1 above)
Comment (shouvSarker) [reply src/main/resources/js/TableFilter.js]: Rebased and pushed!

**Comment by robinMcA** [comment]
Looks good

**Review by robinMcA — APPROVED** [review]

**Pull request merged**